### PR TITLE
[Trivy] Add docker security scan

### PR DESCRIPTION
**Pull request by zbedo**
This PR introduces:

- Docker image security scan, if there are CRITICAL findings, the job will fail.

### CODE REVIEW

**Comment by kyuksel** [comment]
Is there any process/action for non-critical findings?

File: build.sh [hunk 1]
```diff
@@ -33,9 +33,13 @@ docker run --rm  -v $VAULT_LOCATION:/root/.vault-token:ro broadinstitute/dsde-to
 gcloud auth activate-service-account --key-file=image-build-account.json
 gcloud auth configure-docker --quiet
 
-docker image build ./$IMAGE_DIR --tag $GCR_IMAGE_REPO/$IMAGE_DIR:$TAG_NAME --tag $GCR_IMAGE_REPO/$IMAGE_DIR:$VERSION \
-    && docker push $GCR_IMAGE_REPO/$IMAGE_DIR:$TAG_NAME \
-    && docker push $GCR_IMAGE_REPO/$IMAGE_DIR:$VERSION
+docker image build ./$IMAGE_DIR --tag $GCR_IMAGE_REPO/$IMAGE_DIR:$TAG_NAME --tag $GCR_IMAGE_REPO/$IMAGE_DIR:$VERSION 
+
+echo "scanning docker image..."
```
Comment (kyuksel) [inline build.sh:38]: I wonder if we should add a bit more info to the message, like

```suggestion
echo "Scanning the docker image for any critical security vulnerabilities..."
```

**Comment by kyuksel** [comment]
Also curious how long the scan takes

**Comment by zbedo** [comment]
> Also curious how long the scan takes

Usually, it takes a few minutes.

**Review by Qi77Qi — APPROVED** [review]

**Review by kyuksel — APPROVED** [review]
Looks good to me 👍 I assume we'll be ignoring non-critical findings.

**Comment by zbedo** [comment]
> Is there any process/action for non-critical findings?

Docker scans have a lot of vulnerabilities and we cannot afford to fail jobs on every single one of them, so to avoid the noise generated we are skipping any non-critical findings from scanning.

**Comment by adrazhi** [comment]
Are we ignoring `"$HOME"/Library/Caches:/root/.cache/` on a non-MacOS system such as the CI runners? Just making sure.

**Comment by zbedo** [comment]
> Are we ignoring `"$HOME"/Library/Caches:/root/.cache/` on non-MacOS systems such as the CI runners? Just making sure.

@adrazhi Just changed the cache directory path.

**Review by adrazhi — APPROVED** [review]

**Pull request merged**